### PR TITLE
Better primary visual on public pages

### LINKED ISSUE #9933: Download action in share link header split in two
In the header of sharing, on the top right there’s the download actions:
screenshot from 2018-06-20 13-51-10

Now the design for this (and as it worked before) is:
- Both text and 3-dot menu toggle are _one_ element. The text is merely a label for the icon as the actions are all "related" to download (and Download shows in the menu).
- The text "Download" is not displayed on mobile to save space for the title of the file, only the 3-dot-menu remains.

As it currently is:
- It’s split into "header-primary-action" and "header-secondary-action" which is confusing.
- The "Download" text does not vanish on mobile.

Now we could do the following to mix the approaches:
- On mobile, only have the 3-dot icon and all actions in the 3-dot menu (primary action first) like in the intended design
- On bigger sizes, having the primary action outside of the menu (like in the current way)
- To make it less confusing, on desktop we need to make obvious that there’s a primary button, so we should probably add a white border (a la button style) around the text.

What do you think @juliushaertl @skjnldsv @nextcloud/designers?

### PR DESCRIPTION
**Pull request by skjnldsv**
- better visual for the primary button
- hide the download button when the primary is shown

fix #9933
@nextcloud/designers 

@juliushaertl should we always display the primary in the menu and only show it when mobile?
I mean, not only the download? :thinking: 
@nextcloud/designers any thoughts?

### CODE REVIEW

**Comment by jancborchardt** [comment]
(Screenshots please for design changes. ;)

sharing folder page

Looking great! :+1:

**Review by jancborchardt — APPROVED** [review]

**Comment by skjnldsv** [comment]
> should we always display the primary in the menu and only show it when mobile?
I mean, not only the download? :thinking:

What do you think @jancborchardt :)

**Comment by jancborchardt** [comment]
Sorry, I don't really get the question, could you rephrase? :)

It's ok like it is now. A possible change could be to:
- Show the download icon left of the text in the button
- Then on mobile not simply hide the whole button, but only the text so a circular primary download button is still directly shown.

**Comment by skjnldsv** [comment]
I mean, always have the primary action outside AND inside the menu.
But only showing the primary when not in mobile view.

Right now we declare the download action twice and hide the menu via css :/

**Comment by jancborchardt** [comment]
> I mean, always have the primary action outside AND inside the menu.

That doesn’t make any sense though. An action should only be shown once.

> But only showing the primary when not in mobile view.

I don’t get that? Not showing the 3-dot-menu at all?

I’m confused as to what is wrong with the current design:
- On mobile, you see the 3-dot icon with menu: download, direct link, add to your Nextcloud
- On desktop, the important action of "Download" is directly shown, with the others being still in the menu.

This is also what e.g. Material Design does.

**Comment by skjnldsv** [comment]
Sorry, I wasn't clear enough :see_no_evil: 
Currently: the first action is defined as primary and is hidden in mobile view. it will NOT appear in the menu in mobile view. The primary action and the menu are two different things.

I'll suggest we put the primary action back into the menu in mobile mode so we'll have:
1. primary outside and not in menu in desktop view
2. primary not outside BUT in the menu in mobile view.

**Comment by MorrisJobke** [comment]
@skjnldsv Conflicts :/

**Comment by MorrisJobke** [comment]
> @skjnldsv Conflicts :/

Rebase was quite easy 😄

**Review by MorrisJobke — APPROVED** [review]
I like it 👍

**Comment by MorrisJobke** [comment]
> @juliushaertl should we always display the primary in the menu and only show it when mobile?
> I mean, not only the download? 🤔

Let's keep it like that for now in the PR, merge it and discuss about this additional topic in a separate PR/issue.
